src/main/java/org/openrewrite/java/migrate/BeansXmlNamespace.java

```diff
@@ -0,0 +1,89 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.marker.Markers;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.XmlVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class BeansXmlNamespace extends Recipe {
+
+    private static final XPathMatcher BEANS_MATCHER = new XPathMatcher("/beans");
+    private static final String NS_SUN = "http://java.sun.com/xml/ns/javaee";
+    private static final String NS_JCP = "http://xmlns.jcp.org/xml/ns/javaee";
+    private static final String SUN_SCHEMA_LOCATION = "http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/beans_1_0.xsd";
+    private static final String JCP_SCHEMA_LOCATION = "http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd";
+    private static final String DISABLE_VALIDATING = "org.jboss.weld.xml.disableValidating";
+
+    @Override
+    public String getDisplayName() {
+        return "Check valid namespace and schema location in the`beans.xml` file ";
+    }
+
+    @Override
+    public String getDescription() {
+        return " Adds JVM property `org.jboss.weld.xml.disableValidating=true` when valid namespace and schema location is not found in file.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        XmlVisitor<ExecutionContext> xmlVisitor = new XmlVisitor<ExecutionContext>() {
+            @Override
+            public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag t = (Xml.Tag) super.visitTag(tag, ctx);
+                if (!BEANS_MATCHER.matches(getCursor()) || t.getAttributes().stream().map(Xml.Attribute::getKeyAsString).noneMatch("xmlns"::equals)) {
+                    return t;
+                }
+                String nameSpaceValue = null, schemaLocation = null;
+                for (Xml.Attribute attribute : t.getAttributes()) {
+                    if (attribute.getKeyAsString().equals("xmlns")) {
+                        nameSpaceValue = attribute.getValueAsString();
+                    } else if (attribute.getKeyAsString().endsWith("schemaLocation")) {
+                        schemaLocation = attribute.getValueAsString();
+                    } else if (DISABLE_VALIDATING.equalsIgnoreCase(attribute.getKeyAsString())) {
+                        return t;
+                    }
+
+                }
+                if (!checkNameSpaceSchemaLocation(nameSpaceValue, schemaLocation)) {
+                    return addAttribute(t, DISABLE_VALIDATING, String.valueOf(true), ctx);
```

Comment:
The way I read the screenshots attached to this PR `org.jboss.weld.xml.disableValidating` is a JVM property that you pass in at startup, not something to add to `beans.xml`. I could be mistaken as I lack background here, but figured ask before we continue with this PR.

Reply:
you are right @timtebeek  ..i think i understood this wrong..our fix is - ```The automated fix for this rule updates incompatible namespaces with the value appropriate to the specified schemaLocation```.  Will update the PR in a bit